src/pres/Settings.js

```diff
@@ -106,6 +157,7 @@ export default class SettingsView extends React.Component {
                             </span>
                         </Button>
                     </Col>
+                    <OpeningTheory openingManager={this.props.openingManager} />
```

Comment:
This implementation makes an API call every time the board state changes, That is unacceptable for this particular usecase. The implementation should be that we make a call to the API to fetch the url and then open the url AFTER the user clicks "opening theory" button

Reply:
On further review, I think this way makes it so that we know whether to enable/disable the opening theory button. But I still think an API call every is too expensive for this. We can try checking if the page exists, if it does we go to it, otherwise we just google the name of the variation, that we have already calculated.